stm32: test: drivers: memc: ram: fix a specific SRAM memory overflow issue during the build process.

djiatsaf-st opened the pull request:
The D2 domain region of STM32H723/733, STM32H725/735 and STM32H730 has a 32KB memory size, initially separated into two domains.

Due to insufficient memory when building applications with a high footprint, SRAM1 or SRAM2 alone may not be sufficient to build the application.

For example, when building the `tests/drivers/memc/ram ` application on `stm32h735g_disco`, the following error may occur:

```
section `SRAM2' will not fit in region `SRAM2'
region `SRAM2' overflowed by 256 bytes
```

etienne-lms commented on dts/arm/st/h7/stm32h723.dtsi:
```diff
@@ -198,18 +198,11 @@
-		reg = <0x30000000 DT_SIZE_K(16)>;
+		reg = <0x30000000 DT_SIZE_K(32)>;
 		compatible = "zephyr,memory-region", "mmio-sram";
 		zephyr,memory-region = "SRAM1";
```
Rename the region `"SRAM1/2"`?

etienne-lms reviewed (commented):
1 minor suggestion. Otherwise LGTM.

erwango commented on dts/arm/st/h7/stm32h723.dtsi:
```diff
@@ -198,16 +198,9 @@
-	};
-
-	/* D2 domain, AHB SRAM */
-	sram2: memory@30004000 {
```
Removing sram2 will likely create issues on platforms that do not have sram3 when using ethernet. See https://github.com/erwango/zephyr/blob/9cf3f86d32ec22d05d18598514d9dbf8cf76e3f3/soc/st/stm32/stm32h7x/mpu_regions.c#L26.
This should be taken into account, one way or the other.

djiatsaf-st replied:
Add an overlay for stm32h735g_disco

djiatsaf-st committed:
dts: arm: st: h7: combine sram1 and sram2 memory size

Combining the two regions provides enough memory to
successfully build applications.

Note that SRAM1 and SRAM2 are continuous regions.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>
